vms/evm/subscriber.go

```diff
@@ -84,7 +117,87 @@ func (s *subscriber) forwardLogs() {
 			continue
 		}
 		s.log <- *messageInfo
+
+		// Update the database with the latest block height
+		// TODO: This should also be done in a separate goroutine, rather than waiting for warp messages to be processed
```

Comment:
Just to clarify here, do you mean "rather than waiting for the write to disk to finish before processing future messages"?

I definitely agree with the TODOs here though, we should address them prior to merging I think. Another edge case to consider is if there are multiple messages in the same block. We should be careful to only update the "latest block height" once we are sure that all messages prior to (and including?) that block have been processed.

Reply:
Not quite, I more meant we should update the latest processed height on disk periodically, rather than only doing so when we process a warp message. I was thinking we'd want to avoid processing large gaps of blocks _without_ warp messages on startup. Though thinking about it some more, I don't think that's even necessary, since the `eth_getLogs` call on startup won't return any logs for that large gap anyway.

To illustrate, say there's a warp message at block 1000, and the next warp message is at block 2000, but the relayer is offline and unable to process that second message. On restart, whether or not the stored block heigh is 1000 or 1999, the `eth_getLogs` query will return the same result.

Reply:
Good point on updating the latest block height only after all messages have been processed. I'll add that in as part of this PR.

Reply:
So in the current iteration of this PR, we actually store the latest _seen_ block, not the latest fully processed block (I will rename the variables and keys to be more clear). What this means is that if the relayer falls over while processing a large number of warp messages from a particular block, we'll re-process all of those messages on restart. That's obviously not the most efficient use of gas, as already delivered messages would be re-broadcast, so I created this ticket to address that: https://github.com/ava-labs/awm-relayer/issues/22. IMO we can save that for a separate PR, since it doesn't impact correctness here.

Reply:
Yeah all of that reasoning sounds good to me. I just want to confirm that we don't update the latest seen block until after all messages in that block have been processed. Otherwise, if we update the latest seen block prior to processing all messages in that block, we may end up missing messages in that block if the relayer hits a failure and restarts.

Reply:
`eth_getLogs` returns logs from the provided block height _inclusive_. So say we see block 1000, and only process some of the warp messages contained therein. On restart, we'll reprocess block 1000, potentially re processing warp messages, but definitely not missing any.